### DISCUSSION TITLE
Q3S2 (Mid-Aug to Sept) Roadmap Planning

### DISCUSSION THREAD

**Discussion by BenJam** [opencollective/opencollective#4545] [Ideas]
![Slice 1](https://user-images.githubusercontent.com/158833/129197661-e38c2992-5730-42c6-a21b-68ba104efbfb.png)

It's that time of year again... last sprint we updated our priorities slightly, focussing on consolidating on what we have already designed and developed. We anticipated doing this for the remainder of the year and we haven't seen a reason to change that focus so, carry on! 

## Our priorities

> Our priorities do not change, but our framing does. We continue to focus on our core strengths but as a product team, we shall consolidate on our existing success.

2021 has been about deepening our core strengths. It has been great to focus on what makes Open Collective unique, but we have continued to prioritize new features over honing existing experiences that can still limit and frustrate us. Simultaneously 2021 has seen the Open Collective family continue to succeed, particularly in first-party hosts where we're able to leverage our flexibility and our experience to accelerate growth. Experience that we should build upon and share. 

In order to free the team to develop first and third-party hosts, and to focus on the things that are often lost in the heat of feature development, we felt that consolidating our success should be the focus for the remainder of the year. That we should focus on fundamentals like the ledger, completing unfinished projects and, addressing issues that cause ourselves and our user's significant pains and limit our growth.

So what do we _not_ prioritize? We should try not to focus on the needs of smaller hosts, self-hosted collectives, and significant, new feature development. This does not mean that we ignore these needs, and we should continue to listen to our users and understand their concerns, but we should seek to understand their impact on the whole rather than the select few. 

## How do I contribute?

Due to the nature of the priorities above you may have in mind a list of smaller issues that you feel are particularly impactful and aligned with our priorities. 

That's fine. 

If you're able to group issues and problems into 'themes' this will be helpful as we plan the next six weeks. Otherwise, we will work to organize issues raised along broad themes so that we don't spend the next six weeks to context-switching and discussing minutiae. 

#### Proposing issues
Create a top-level comment on this discussion, link to any issues that already exist in this repo (a quick search is good enough) but don't worry if you don't have the time or can't find anything. Content is more important than linking.

#### Scoping the problem
If you feel that the idea for a project or a specific issue isn't well-scoped then feel free to loop in others to help scope it down. This might involved validating the problem with some quantitative research, asking someone whether they've encountered anyone else with a similar problem or reaching out to users specifically to check an idea or validate an assumption. There is no need to scope the solution at this stage, we are looking for the most impactful problems to solve that are in line with our mission, our values and our current focus. 

#### Discussion
Feel free to lend thoughts, opionions, data and ask questions in threads from each of the top-level comments (which should contain project).

#### Voting
Feel free to +1 any project ideas by hitting the ⬆️  button alongside any top-level comment. Try to limit yourself to three votes for the sprint to keep things focussed.

#### Synchronous discussion
We'll arrange a synchronous discussion for everyone **Thursday 19th August** to talk through the top pitches for the quarter and to create the space for us to decide together based on the outcome of this discussion.

#### Kickoffs and further discussion
Depending on the level of shared understanding between product, design and engineering we may not need a kick-off for each individual issue that that issue contains the necessary detail. A quick 1:1 1:many discussion to scope the issue(s) within a theme may get us through.

**Comment by znarf** [comment]
### Credit Card Fraud

- Implement a Captcha system for guest contributions (Recaptcha or something else)
- Implement some extra parameters from Frontend to help classify Orders

**Reply by BenJam** [reply]
I created an issue for this a while back, so adding here https://github.com/opencollective/opencollective/issues/4329

**Comment by znarf** [comment]
### Financial Contributions between Fiscal Hosts

https://github.com/opencollective/opencollective/issues/3190

We're confident we have now a solid foundation to make this happen. This is asked by Google Season of Docs for 2022.

**Reply by alanna** [reply]
Big +1 to this one

**Reply by piamancini** [reply]
+1 : Just run into this issue again

**Comment by znarf** [comment]
### New CSVs

Finalize the new CSV format and ship it everywhere across the platform.

**Reply by BenJam** [reply]
using this issue to cover this https://github.com/opencollective/opencollective/issues/4426

**Comment by BenJam** [comment]
### Open Source Collective Automatic Approval Adjustments 

Our automatic 👍ing of projects onto Open Source Collective does not match our eligibility criteria. We're going to take a look at the criteria this month (most likely mandating two _maintianers_ and that a project isn't owned by a user-level account. All for the sake of bus factor etc. 

I'd like to make sure that the automatic process matches our manual review processes. Open Source Collective will pay for this obvs.

**Comment by kewitz** [comment]
### Merge settings and host dashboard

https://github.com/opencollective/opencollective/issues/3880

**Reply by BenJam** [reply]
covered in @Memo-Es's proposl below so treating that as a +1 :)

**Comment by alanna** [comment]
## Improve Events

Events continue to be a feature users want. There are quite a few low-hanging fruit for improvements that would make a difference. The world is opening up and events will start happening more, and we're seeing more high-profile online events on the platform as well.

#4280

**Comment by alanna** [comment]
## Integrate with accounting tool & practices

This projects lists a lot of issues we keep running into.

#4396

**Reply by BenJam** [reply]
I agree that this is something to invest in in the future, but I kinda feel like the eCSV work that @znarf et al have prioritised (which we'll be continuing) is a good first step here.

**Comment by alanna** [comment]
## Privacy on a transparent platform

We have known about challenges in this area for a while, and it's also coming up in @piamancini 's recent work identifying a bunch of incognito/guest contributions. We need to give users better options for representing themselves publicly which also meet our needs around reporting.

#4214 - I think the most important issue is #3689, and once we have that we can tackle #3422 and #3340

**Reply by Betree** [reply]
https://github.com/opencollective/opencollective/issues/3689 sounds simple enough and seems quite important, I'm all in to prioritize it this sprint.

The other two are more complex, so maybe we can keep them for later.

**Reply by piamancini** [reply]
agreed.

**Comment by alanna** [comment]
## Attach payable invoice to bank transfer contribution emails

Seems like it would be simple. This would help large hosts because contributors needing invoices could generate them automatically.

#4497

**Comment by alanna** [comment]
## Calculate host fee on pre-tax total

This is creating an unfair situation for Collectives because they are losing money to host fees they shouldn't have to pay.

#4492

**Comment by alanna** [comment]
## Reveal total payout sum per payee to host admin

OCF has a heavy compliance burden and this would help a lot.

#4491

**Reply by Betree** [reply]
This would also help for https://github.com/opencollective/opencollective/issues/3258

**Reply by BenJam** [reply]
Hmm worth considering this from the perspective that Open Collective Foundation is a big Open Collective contributor now...

**Comment by alanna** [comment]
## Make reply to email for team addition notification the adder only

We need to make small tweaks like this to reduce support load as it's stretched pretty thin.

#4461

**Comment by alanna** [comment]
## Represent funds coming through Events and Projects in the parent Collective budget stats 

Not doing this creates a misleading picture of the parent Collective's finances and activity level.

#4360

**Reply by phoresced** [reply]
Preferably this would be a setting, so project teams who decide to maintain budget autonomy wouldn't have their budget included in the parent Collective.

**Comment by alanna** [comment]
## Invite organization to submit an expense

We're experiencing pain when payee profiles and actual payees don't match up. One way to reduce this happening is to enable people more experienced with the platform to invite an expense with the correct details pre-filled. Ensuring the expense comes from the payee org a not the individual invitee is probably the most common need in this area, and this is one tool for it.

#4224

**Comment by alanna** [comment]
## Remove the Host Plan selection settings 

I consider this to be a hangover to-do from the new pricing model project. it would be great to complete it.

#4056

**Comment by alanna** [comment]
## Host admin can define more manual payment methods 

Considering we can implement this using a simplified version of the bank account payment method setup we already have (no structured account data, just a plain instructions text box, same email notification pattern and thankyou screen data display), we should do it. I think the only real new part is enabling the user to set a custom name for the payout method. OCF would like to use this to set up a cheque option.

#4024

**Comment by alanna** [comment]
## Do not treat system-generated expense summary as an attachment 

Small front-end change that would really help host admins.

#3865

**Comment by alanna** [comment]
## Host can require minimum number of admins 

Another one that would reduce onboarding burden on OCF support staff, and I believe @BenJam was interested in it as well for OSC.

#3593

**Reply by BenJam** [reply]
I'm putting a +1 hat on this as we spend a lot of time explaining this.

**Comment by alanna** [comment]
## Reducing Stripe fees on small transactions 

If there is any easy way to try to get a deal with Stripe, I think we should try. This issue disproportionately affects small Collectives and disincentivizes people from supporting lots of projects with modest amounts.

#3542

**Comment by alanna** [comment]
## Host admin can unhost Collective 

This is an issue just waiting to blow up someday. Hosts have to be able to kick out Collectives.

#3330

**Comment by Betree** [comment]
# PayPal synchronization issues

https://github.com/opencollective/opencollective/issues/4368

I'm still carrying some work from this project due to new issues being discovered along the road and my limited availability because of the holidays. Like previous sprint, it's important to prioritize this to avoid things that will later be difficult to fix in support.

**Comment by alanna** [comment]
## Host admin can export Collective data CSV 

OCF/OSC have been heavily leaning on metabase to extract this kind of data, but other hosts don't have such access. Considering how much we're using that access I believe it would be very valuable to other hosts as well and could be achieved through a CSV export with minimal design.

#2808

**Reply by piamancini** [reply]
+1

**Reply by BenJam** [reply]
yupyupyup

**Reply by fernandasca** [reply]
+1

**Comment by alanna** [comment]
## Set up redirect URL for host terms 

I just really really want this. It comes up multiple times a week for me, sometimes multiple times a day. Is it really so hard to set up a redirect? I can't understand what happened in that thread.

#2757

**Comment by alanna** [comment]
## Customizable "thank you" email to contributors 

We want to support our Collectives to engage with their communities, and this is an easy win in my mind. If we could enable it at the level of Events and Projects too, that would be amazing.

#2682

**Reply by Betree** [reply]
This one would be about creating specs, there's not enough material atm to start the implementation

**Comment by alanna** [comment]
## Unable to pay 100% of collective balance due to fees 

This continues to cause problems. There must be some solution.

#1073

**Comment by alanna** [comment]
## Ability for hosts to set a monthly flat host fee per Collective instead of a % 

Requested by multiple hosts.

#3362

**Comment by alanna** [comment]
## Manually subscribe to Updates 

On the theme of enabling community engagement. 

#1445

**Comment by jfdh** [comment]
### New payment methods

Add new payment methods that are already supported by Stripe for each region (i.e. Bancontact for Belgium - iDeal for NL)

https://stripe.com/docs/payments/payment-methods/overview

#4327

**Reply by sunnyleen** [reply]
Yes!! And with a qr-code for direct bank transfers!

**Comment by jfdh** [comment]
### Disable automatic emails for host admins

Give the possibility to disable emails such as "New pending financial contribution" and disable all automatic emails if you're admin of a host but don't need to be kept up to date with every movement of the host.

**Reply by piamancini** [reply]
Agreed! this makes it almost impossible for host admins to see the notifications they care about. Like I need to see expense notifications for 5 collectives only not every single one.

**Comment by piamancini** [comment]
## Apply the guest contribution flow parameters 

$5000 address requirement, $250 email / name requirement to all payment flows, included added funds.

**Reply by Betree** [reply]
We should at least focus on specs for this one, especially since some design will be needed for the contribution flow and add funds modal.

For the contribution flow, https://github.com/opencollective/opencollective/issues/3478 could be tackled at the same time since we'll need this info in the frontend to properly check the amount.

**Reply by kewitz** [reply]
Related: https://github.com/opencollective/opencollective/issues/4498

**Comment by piamancini** [comment]
## Ship Open Stocks 

(issue coming up)

**Reply by piamancini** [reply]
Issue: https://github.com/opencollective/opencollective/issues/4557

**Comment by alanna** [comment]
## Bugs & Papercuts

Squash as many small issues and simple fixes as possible. 

* [Minimal complexity](https://github.com/opencollective/opencollective/issues?q=is%3Aopen+is%3Aissue+label%3A%22complexity+%E2%86%92+minimal%22)
* [Papercuts](https://github.com/opencollective/opencollective/issues?q=is%3Aopen+is%3Aissue+label%3Apapercuts)

**Comment by alanna** [comment]
## Facilitate Bounties

We've had some really good bounty contributions recently. If we spent some time scoping [bounty candidate](https://github.com/opencollective/opencollective/issues?q=is%3Aopen+is%3Aissue+label%3A%22bounty+candidate%22) issues, maybe more would get done.

**Comment by piamancini** [comment]
Implement the Budget Page (Reports) for hosts at least.

**Reply by piamancini** [reply]
https://github.com/opencollective/opencollective/issues/4300

**Reply by piamancini** [reply]
Is it related to what @znarf is proposing above for CSV?

**Reply by znarf** [reply]
No, my CSV is proposal is CSV only.

**Comment by znarf** [comment]
### GDPR consent

As this issue is on our radar, we have a duty to prioritize it.

https://github.com/opencollective/opencollective/issues/4537

**Comment by Memo-Es** [comment]
## Implement design-ready projects

We have a bucket list of projects ready to be implemented. Earlier this week, we had a chat about aligning that bucket list with our strategic priorities this year, and we have some agreement on what to implement next:

 - Admin panel update (aka merge settings and host dashboard) #3880 
 - Reports section for hosts #4300 also mentioned in https://github.com/opencollective/opencollective/discussions/4545#discussioncomment-1205867
 - Onboarding updates
   - Hosts https://github.com/opencollective/opencollective/issues/3758
   - Collectives https://github.com/opencollective/opencollective/issues/4542

**Comment by Memo-Es** [comment]
## Marketing Update (Continuation)

In the last sprint, we started dedicating the design resources to marketing and less on features, at least for what's left of 2021. We made the progress of bringing the marketing pages we had left to the updated design. The "How it works?" page and "Help and Support" pages are ready along with the navbar update, so for this sprint, we propose to continue with the following efforts:

**Implementation**
 - Implement the how it works and help and support page
 - Implement the 404, unknown error, under construction and page not found pages
 - Implement the navbar update

**Design**
- Homepage update. Showcase OC as an org and display its initiatives and services
- Online sharing. Standardize how we display OC content online (Twitter, thumbnails, etc.)
- Developers website. We talked about this in the past but dropped the ball at some point, so I think it would be good to re-pick it.

**Reply by alanna** [reply]
I agree about implementing the designs that are already done but I'm not sure about the rest. I think we should hire a marketing person and let them lead this area before running too far ahead.

**Reply by Memo-Es** [reply]
Totally on board with having a marketing person hired. Until that happens, we can work on having a decent basis for building a real marketing strategy, and I don't think we are very far from having that. What we built as the OC homepage a year ago should not actually be the homepage, and IMO is one of the main sources of confusion when interacting with what's OC.

**Reply by piamancini** [reply]
I agree with design starting to re think the homepage to be more clarifying and represent better the platform + 1st party hosts proposal. Right now it's not there yet. 

Also, there's a big design backlog that engineering is not able to process yet, so we don't want to do anymore design for the moment. Marketing, homepage, etc. seems like the right focus. When we have a marketing person, they can build on what we have, but it's not going to happen in the next 6 weeks. 

Design can also support OCF now with some specific needs it has.
